src/foundry/common/data/data.mjs/tokenData.d.ts

```diff
@@ -0,0 +1,540 @@
+   * @defaultValue `[]`
+   */
+  effects?: string[] | null;
+
+  /**
+   * A single icon path which is displayed as an overlay on the Token
+   */
+  overlayEffect?: string | null;
+
+  /**
+   * @defaultValue `1`
+   */
+  alpha?: number | null;
+
+  /**
+   * Is the Token currently hidden from player view?
+   * @defaultValue `false`
+   */
+  hidden?: boolean | null;
+
+  /**
+   * Is this Token a source of vision?
+   * @defaultValue Whether dimSight or brightSight have a value greater 0.
+   */
+  vision?: boolean | null;
+
+  /**
+   * How far in distance units the Token can naturally see as if in dim light
+   * @defaultValue `0`
+   */
+  dimSight?: number | null;
+
+  /**
+   * How far in distance units the Token can naturally see as if in bright light
+   * @defaultValue `0`
+   */
+  brightSight?: number | null;
+
+  /**
+   * How far in distance units this Token emits dim light
+   * @defaultValue `0`
+   */
+  dimLight?: number | null;
+
+  /**
+   * How far in distance units this Token emits bright light
+   * @defaultValue `0`
+   */
+  brightLight?: number | null;
+
+  /**
+   * The angle at which this Token is able to see, if it has vision
+   * @defaultValue `360`
+   */
+  sightAngle?: number | null;
+
+  /**
+   * The angle at which this Token is able to emit light
+   * @defaultValue `360`
+   */
+  lightAngle?: number | null;
+
+  /**
+   * @defaultValue `0.25`
+   */
+  lightAlpha?: number | null;
+
+  /**
+   * A data object which configures token light animation settings
+   * @defaultValue `{}`
+   */
+  lightAnimation?: AnimationDataConstructorData | null;
+
+  /**
+   * A displayed Token disposition from CONST.TOKEN_DISPOSITIONS
+   * @defaultValue `CONST.TOKEN_DISPOSITIONS.HOSTILE`
+   */
+  disposition?: CONST.TokenDisposition | null;
+
+  /**
+   * The display mode of Token resource bars, from CONST.TOKEN_DISPLAY_MODES
+   * @defaultValue `CONST.TOKEN_DISPLAY_MODES.NONE`
+   */
+  displayBars?: CONST.TokenDisplayMode | null;
+
+  /**
+   * The configuration of the Token's primary resource bar
+   * @defaultValue `{ attribute: game?.system.data.primaryTokenAttribute || null }`
+   */
+  bar1?: TokenBarDataConstructorData | null;
+
+  /**
+   * The configuration of the Token's secondary resource bar
+   * @defaultValue `{ attribute: game?.system.data.secondaryTokenAttribute || null }`
+   */
+  bar2?: TokenBarDataConstructorData | null;
+
+  /**
+   * An object of optional key/value flags
+   * @defaultValue `{}`
+   */
+  flags?: Record<string, unknown> | null;
+}
+
+/**
+ * The data schema for a Token document.
+ */
+export class TokenData extends DocumentData<
+  TokenDataSchema,
+  TokenDataProperties,
+  PropertiesToSource<TokenDataProperties>,
+  TokenDataConstructorData,
+  documents.BaseToken
+> {
+  static defineSchema(): TokenDataSchema;
+
+  /**
+   * The default icon used for newly created Item documents
+   */
+  static DEFAULT_ICON: typeof CONST.DEFAULT_TOKEN;
```

Comment:
I'd make the type `string` and document `CONST.DEFAULT_TOKEN` as the default value so that users can override / change this if they want to.